### ISSUE TITLE
Ignore commented lines?

### ISSUE THREAD

**Issue by eyefodder** [conventionalcommit/commitlint#7]
Is it possible to get commitlint to ignore commented lines? For example this commit message:
```
ops: update CommitLint allowed types
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch DAY-36-enforce-git-hygeine
# Your branch is up to date with 'origin/DAY-36-enforce-git-hygeine'.
#
# Changes to be committed:
#	modified:   .commitlint.yml
#
```

will fail because there's not an empty line after `ops: update CommitLint allowed types` but given the rest of the lines are all commented out, I think it should pass? I'm also finding the same thing if e.g. some of the commented lines are longer than the rules (this often happens if the path to a modified file is long)

Cheers!

**Comment by danbernier** [comment]
I'd add to @eyefodder's point that anything that `git commit -v` ignores, so should commitlint.

**Comment by gumbo2k** [comment]
Yes, it would be lovely to have the `git commit`'s [cleanup of comments and verbose diff](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---cleanupltmodegt) happen before commitlint sees the message. 

This would shift the burden of maintaining the filtering of comments and diff's (in case of `git commit -v`) to git itself, instead of trying to emulate that (configurable) behavior. :thinking: 

Unfortunately there seems to be no `postprocess-commit-msg` hook in the [Committing-Workflow Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) that we could use.

**Comment by gumbo2k** [comment]
BTW: The [conventional-changelog 's commit-lint](https://github.com/conventional-changelog/commitlint) also seems to do this job only way ... ignoring comments by [hardcoding the comment character as `#`](https://github.com/conventional-changelog/conventional-changelog/pull/231/commits/e2a45f3d0e51586413d2238a071cf7aa3e013ed6) but [failing on the diffs that a `git commit --verbose` will put in the commit message](https://github.com/conventional-changelog/commitlint/issues/437).
